source/common/router/route_config_update_receiver_impl.cc

```diff
@@ -18,47 +18,73 @@ bool RouteConfigUpdateReceiverImpl::onRdsUpdate(const envoy::api::v2::RouteConfi
   if (new_hash == last_config_hash_) {
     return false;
   }
-
   route_config_proto_ = rc;
   last_config_hash_ = new_hash;
+  const uint64_t new_vhds_config_hash = rc.has_vhds() ? MessageUtil::hash(rc.vhds()) : 0ul;
+  vhds_configuration_changed_ = new_vhds_config_hash != last_vhds_config_hash_;
```

Comment:
My concern is the ABA problem on `vhds_configuration_changed_` because `vhds_configuration_changed_` is overrided instead of stacked.
But I don't have a clear view the usage of vhds_configuration_changed_ is correct or not

Reply:
What is the ABA problem?

Reply:
I believe @lambdai was concerned with lost updates. This was raised at the same time with the thread here; https://github.com/envoyproxy/envoy/pull/8617/files/01f970263a8a203743e669af037c6b095f7dcd92#diff-dcbe1eb34bb1b29e349d7c87c0f20337R57

Left a TODO notice in the code.